src/PDPVerifier.sol

```diff
@@ -37,13 +37,19 @@ contract PDPVerifier is Initializable, UUPSUpgradeable, OwnableUpgradeable {
     uint256 public constant NO_CHALLENGE_SCHEDULED = 0;
 
     // Events
-    event ProofSetCreated(uint256 indexed setId);
+    event ProofSetCreated(uint256 indexed setId, address indexed owner);
+    event ProofSetOwnerChanged(uint256 indexed setId, address indexed oldOwner, address indexed newOwner);
     event ProofSetDeleted(uint256 indexed setId, uint256 deletedLeafCount);
-    event RootsAdded(uint256 indexed firstAdded);
-    event RootsRemoved(uint256[] indexed rootIds);
-    event ProofFeePaid(uint256 indexed setId, uint256 fee, uint64 price, int32 expo);
     event ProofSetEmpty(uint256 indexed setId);
 
+    event RootsAdded(uint256 indexed setId, uint256[] rootIds);
+    event RootsRemoved(uint256 indexed setId, uint256[] rootIds);
+   
+    event ProofFeePaid(uint256 indexed setId, uint256 fee, uint64 price, int32 expo);
+
+    event PossessionProven(uint256 indexed setId, uint256 challengedLeafCount, uint256 seed, uint256 challengeCount);
```

Comment:
I think what we really want is actually a list of (root indexes, root offsets) that are challenged: 
```solidity
event PossessionProven(uint256 indexed setId, RootIdAndOffset[] challenges)
```

Reply:
Fixed.